src/stats.rs

```diff
@@ -0,0 +1,135 @@
+use crate::config::{version, Config};
+use crate::download::cache_info_with_warnings;
+use crate::util::repo_aur_pkgs;
+
+use alpm::PackageReason;
+use alpm_utils::DbListExt;
+
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+use anyhow::Result;
+use indicatif::HumanBytes;
+
+struct Info<'a> {
+    total_packages: usize,
+    repo_packages: usize,
+    aur_packages: usize,
+    explicit_packages: usize,
+    total_size: i64,
+    max_packages: Vec<(i64, &'a str)>,
+}
+
+async fn collect_info<'a>(config: &'a Config, max_n: usize) -> Result<Info<'a>> {
+    let db = config.alpm.localdb();
+    let sync_db = config.alpm.syncdbs();
+
+    let total_packages = db.pkgs().len();
+    let mut foreign_packages = Vec::new();
+    let mut explicit_packages = 0;
+    let mut total_size = 0;
+
+    let mut max_packages = BinaryHeap::with_capacity(max_n + 1);
+
+    for pkg in db.pkgs() {
+        max_packages.push(Reverse((pkg.isize(), pkg.name())));
+        if max_packages.len() > 10 {
+            max_packages.pop();
+        }
+        if let Err(alpm::Error::PkgNotFound) = sync_db.pkg(pkg.name()) {
+            foreign_packages.push(pkg.name());
+        }
+        if pkg.reason() == PackageReason::Explicit {
+            explicit_packages += 1;
+        }
+        total_size += pkg.isize();
+    }
+
+    let (repo, aur) = repo_aur_pkgs(config);
```

Comment:
aur is just the possible aur packages. When you do cache_info_with_warnings, warnings.pkgs.len() is the real amount of aur packages.